controllers/vrg_volrep.go

```diff
@@ -1466,26 +1466,47 @@ func setPVCClusterDataProtectedCondition(protectedPVC *ramendrv1alpha1.Protected
 	}
 }
 
-// deleteVR deletes a VolumeReplication instance if found
-func (v *VRGInstance) deleteVR(vrNamespacedName types.NamespacedName, log logr.Logger) error {
+// ensureVRDeletedFromAPIServer adds an additional step to ensure that we wait for volumereplication deletion
+// from API server before moving ahead with vrg finalizer removal.
+func (v *VRGInstance) ensureVRDeletedFromAPIServer(vrNamespacedName types.NamespacedName, log logr.Logger) error {
 	cr := &volrep.VolumeReplication{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      vrNamespacedName.Name,
 			Namespace: vrNamespacedName.Namespace,
 		},
 	}
 
-	err := v.reconciler.Delete(v.ctx, cr)
+	err := v.reconciler.APIReader.Get(v.ctx, vrNamespacedName, cr)
 	if err == nil || k8serrors.IsNotFound(err) {
 		return nil
 	}
 
-	log.Error(err, "Failed to delete VolumeReplication resource")
+	log.Info("Wait to delete VolumeReplication resource from API server", "Error", err)
```

Comment:
Reword message "Waiting for VolumeReplication resource to be deleted from the API server"